tailnet/conn.go

```diff
@@ -569,54 +582,60 @@ func (c *Conn) BlockEndpoints() bool {
 	return c.blockEndpoints
 }
 
-// AwaitReachable pings the provided IP continually until the
-// address is reachable. It's the callers responsibility to provide
-// a timeout, otherwise this function will block forever.
-func (c *Conn) AwaitReachable(ctx context.Context, ip netip.Addr) bool {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel() // Cancel all pending pings on exit.
-
-	completedCtx, completed := context.WithCancel(context.Background())
-	defer completed()
-
-	run := func() {
-		// Safety timeout, initially we'll have around 10-20 goroutines
-		// running in parallel. The exponential backoff will converge
-		// around ~1 ping / 30s, this means we'll have around 10-20
-		// goroutines pending towards the end as well.
-		ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
-		defer cancel()
-
-		_, _, _, err := c.Ping(ctx, ip)
-		if err == nil {
-			completed()
-		}
-	}
-
-	eb := backoff.NewExponentialBackOff()
-	eb.MaxElapsedTime = 0
-	eb.InitialInterval = 50 * time.Millisecond
-	eb.MaxInterval = 30 * time.Second
-	// Consume the first interval since
-	// we'll fire off a ping immediately.
-	_ = eb.NextBackOff()
+var unixEpoch = time.Unix(0, 0)
 
-	t := backoff.NewTicker(eb)
-	defer t.Stop()
+// AwaitReachable waits until the peer with given IP is reachable, which we define as having handshaked at least once at
+// the Wireguard layer. It's the callers responsibility to provide  a timeout, otherwise this function will block
+// forever.
+func (c *Conn) AwaitReachable(ctx context.Context, ip netip.Addr) bool {
+	rightAway := make(chan struct{}, 1)
+	rightAway <- struct{}{}
+	tkr := time.NewTicker(50 * time.Millisecond)
```

Comment:
Is 50ms a bit high to start off with? In non-ideal conditions every connection attempt may have an artificial delay of 50ms. A backoff could still be useful here.

Reply:
50ms was the initial interval from the ping-based code, so I just kept it.  What do you think is more reasonable?

Reply:
That’s fair. And good question. I suppose something in the 15-25ms range would be nice. But I’m mostly basing this on what I feel a fast non-local ping is like.